### ISSUE TITLE
DropdownMenuItem: Add new prop role.

### ISSUE THREAD

**Issue by olehnoskov-cengage** [cengage/react-magma#2136]
Adopters need a new prop `role` for the `DropdownMenuItem` component to improve accessibility.
At this time, we have a default `menuitem` role for `DropdownMenuItem`.

AC:
Add a new prop `role` for the `DropdownMenuItem`.

[Original request](https://cengage.slack.com/archives/C8Y4FR63U/p1765801140645409)

We need to check the discussion about this issue, following the link above.
